### PR TITLE
Remove Useless Security Checks.

### PR DESCRIPTION
**Pull request by vancem**
CoreClr does not provide code access security features.  

However there is (expensive) code that is still being run associated with this feature.   This change #if defs this out for CoreCLR (we don't delete it because the code is still shared with desktop).   

Note that there may be more that can be ripped out.  I only did the stuff that was relatively easy to find.   There may also be dead code that has not be eliminated

### CODE REVIEW

**Comment by vancem** [comment]
@stephentoub

File: src/mscorlib/src/System/RtType.cs [hunk 1]
```diff
@@ -5466,9 +5466,11 @@ internal Object CreateInstanceDefaultCtor(bool publicOnly, bool skipCheckThis, b
 
                     if (ace.m_ctor != null)
                     {
+#if !FEATURE_CORECLR
                         // Perform security checks if needed
                         if (ace.m_bNeedSecurityCheck)
                             RuntimeMethodHandle.PerformSecurityCheck(instance, ace.m_hCtorMethodHandle, this, (uint)INVOCATION_FLAGS.INVOCATION_FLAGS_CONSTRUCTOR_INVOKE);
```
Comment (jkotas) [inline src/mscorlib/src/System/RtType.cs]: This method is fast path. We should also ifdef out the security checks on the regular path in `RuntimeTypeHandle::CreateInstance`.

**Comment by stephentoub** [comment]
FYI, I ran corefx's System.Runtime, System.Reflection, System.Reflection.Emit, System.Dynamic.Runtime, and System.Linq.Expressions tests with this on Windows, and they all passed.

File: src/mscorlib/src/System/RtType.cs (hunk 1 above)
Comment (vancem) [inline src/mscorlib/src/System/RtType.cs]: @jkotas.   I have added the bSecurityCheckOff = true to the CreateIntanceSlow to address this.   There are actually MANY places where you can try to turn off security, on this code path, so maybe we should put it elsewhere (e.g. in the body of RuntimeTypeHandle.CreateInstance), but making the change there seemed to make complex logic MORE complex, so I chose this for simplicity.   Let me know if you prefer it elsewhere.

**Comment by jkotas** [comment]
LGTM. Could you please squash into one commit?

**Comment by jkotas** [comment]
Superseded by #3027

**Pull request closed without merging**